packages/style-loader/index.js

```diff
@@ -1,10 +1,66 @@
-module.exports = ({ config }, options = {}) => config.module
-  .rule(options.ruleId || 'style')
-    .test(/\.css$/)
-      .use(options.styleUseId || 'style')
-        .loader(require.resolve('style-loader'))
-        .when(options.style, use => use.options(options.style))
-        .end()
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const merge = require('deepmerge');
+
+module.exports = (neutrino, opts = {}) => {
+  const isBuild = neutrino.options.command === 'start';
+  const options = merge({
+    styleUseId: 'style',
+    hot: true,
+    extract: {
+      plugin: {
+        filename: isBuild ? '[name].css' : '[name].[contenthash].css'
+      }
+    }
+  }, opts);
+
+  neutrino.config.module
+    .rule(options.ruleId || 'style')
+      .test(/\.css$/)
+      .use(options.styleUseId)
+         .loader(require.resolve('style-loader'))
+         .when(options.style, use => use.options(options.style))
+         .end()
       .use(options.cssUseId || 'css')
         .loader(require.resolve('css-loader'))
         .when(options.css, use => use.options(options.css));
+
+  if (options.extract) {
+    const styleRule = neutrino.config.module.rule('style');
+    const styleEntries = styleRule.uses.entries();
+    const useKeys = Object.keys(styleEntries).filter(key => key !== options.styleUseId);
+
+    options.extract.loader = Object.assign({
+      use: useKeys.map(key => ({
+        loader: styleEntries[key].get('loader'),
+        options: styleEntries[key].get('options')
+      })),
+      fallback: {
+        loader: styleEntries[options.styleUseId].get('loader'),
+        options: styleEntries[options.styleUseId].get('options')
+      }
+    }, options.extract.loader || {});
+
+    styleRule
+      .uses
+        .clear()
+        .end()
+      .when(options.hot, (rule) => {
+        rule.use(options.cssHotLoaderUseId || 'cssHotLoader')
```

Comment:
Yeah, my comment before wasn't very clear, my apologies. I was asking to name this ID from `cssHotLoader` to just `hot`, just the HotPlugin does in other middleware.